include/seqan3/io/sequence_file/format_fasta.hpp

```diff
@@ -329,47 +326,31 @@ class format_fasta
         if (options.fasta_blank_before_id)
             stream_it = ' ';
 
-        std::ranges::copy(id, stream_it);
+        stream_it.write_range(id);
 
         detail::write_eol(stream_it, options.add_carriage_return);
     }
 
     //!\brief Implementation of writing the sequence.
-    template <typename stream_it_t,
-              typename seq_type>
-    void write_seq(stream_it_t & stream_it,
-                   sequence_file_output_options const & options,
-                   seq_type && seq)
+    template <typename stream_it_t, typename seq_type>
+    void write_seq(stream_it_t & stream_it, sequence_file_output_options const & options, seq_type && seq)
     {
+        auto char_sequence = seq | views::to_char;
+
         if (options.fasta_letters_per_line > 0)
         {
-        #if SEQAN3_WORKAROUND_VIEW_PERFORMANCE
-            size_t count = 0;
-            for (auto c : seq)
+            auto it = std::ranges::begin(char_sequence);
+
+            while (it != std::ranges::end(char_sequence))
             {
-                stream_it = to_char(c);
-                if (++count % options.fasta_letters_per_line == 0)
-                    detail::write_eol(stream_it, options.add_carriage_return);
-            }
-            if (count % options.fasta_letters_per_line != 0)
+                auto end_it = std::ranges::next(it, options.fasta_letters_per_line, std::ranges::end(char_sequence));
```

Comment:
Here at least, you only need to move by `options.fasta_letters_per_line`, above the amount increases on every iteration.

Reply:
Well yes, there is no way I can circumcent this, except if I write char by char or if I introduce a `write_n(it, n, sentinal)` function to the iterator.. I thought you decided against both?

Reply:
This was just explanation for the previous comment, this if fine.

> I introduce a write_n(it, n, sentinal) function to the iterator.. I thought you decided against both?

As I said before, the `write_n` function won't help us. It makes this potentially faster but the case where the size is not known potentially slower (because of requiring `distance`).

Reply:
[except that this should be replaces with the `couning_next` that I proposed.